src/api/app/models/bs_request/find_for/group.rb

```diff
@@ -4,24 +4,39 @@ class Group < Base
       include UserGroupMixin
 
       def all
-        inner_or = []
-        group = ::Group.find_by_title!(group_title)
+        query = union_query
+        @relation = if query.present?
+                      @relation.where("bs_requests.id IN (#{query})")
+                    else
+                      @relation.none
+                    end
+      end
+
+      def all_count
+        ActiveRecord::Base.connection.execute("SELECT COUNT(bs_request_id) FROM (#{union_query}) x").first.first
+      end
+
+      private
 
-        # find requests where group is maintainer in target project
-        @relation, inner_or = extend_query_for_maintainer(group, @relation, roles, inner_or)
+      def group
+        @group ||= ::Group.find_by_title!(group_title)
+      end
+
+      def union_query
+        query_parts = []
+        query_parts << bs_request_actions_query.to_sql if bs_request_actions_query
+        query_parts << reviews_query.to_sql if reviews_query
+        query_parts.join(' UNION ')
```

Comment:
[active_record_union](https://github.com/brianhempel/active_record_union) looks pretty active to me.

Reply:
We checked it, but we thought it was not worthwhile to add a gem to use it just once. In addition, I am not sure if `union(nil)` works. Otherwise we will have to keep the `ifs` anyway. I can still give it a try.

Reply:
the gem is not working for this case: https://github.com/brianhempel/active_record_union/issues/23